src/js/stores/library.js

```diff
@@ -26,9 +26,110 @@ define(function (require, exports, module) {
 
     var Fluxxor = require("fluxxor"),
         Immutable = require("immutable"),
+        EventEmitter = require("eventEmitter"),
         _ = require("lodash");
 
-    var events = require("../events");
+    var events = require("../events"),
+        log = require("js/util/log");
+    
+    /**
+     * LibrarySyncStatus is a wrapper class of the AdobeLibraryCollection instance to provide an easy way
+     * for listening to its sync status. Read LibrarySyncStatus#addSyncListener for more details of the 
+     * sync status.
+     */
+    var LibrarySyncStatus = function (libraryCollection) {
+        this._libraryCollection = libraryCollection;
+        this._syncingLibraryIDs = new Set();
+        this._emitter = new EventEmitter(this);
+        
+        this._status = {
+            isSyncing: false,
+            libraryNumber: this.getFilteredLibraries().length
+        };
+        
+        this._libraryCollection.addSyncListener(this._onLibraryCollectionSync.bind(this));
+    };
+    
+    /**
+     * Listen to libraries' sync status. 
+     *
+     * @param {function(isSyncing, libraryNumberChanged)} listener
+     *        isSyncing - if true, one or more libraries are uploading or downloading.
+     *        libraryNumberChanged - if true, the number of the libraries has increased or decresaed. 
+     * 
+     */
+    LibrarySyncStatus.prototype.addSyncListener = function (listener) {
+        this._emitter.addListener("sync", listener);
+    };
+    
+    /**
+     * Handle AdobeLibraryCollection sync event.
+     *
+     * @private
+     */
+    LibrarySyncStatus.prototype._onLibraryCollectionSync = function () {
+        log.debug("[CC Lib] on collection change");
+        
+        var newLibraries = this._libraryCollection.getNewLibrarySyncProgress();
+        
+        newLibraries.forEach(function (data) {
```

Comment:
It would help a lot to describe what each block is doing in this method, it seems central to the whole syncing thing.